modules/charon-core/src/main/java/org/wso2/charon3/core/objects/RoleV2.java

```diff
@@ -0,0 +1,649 @@
+            this.attributeList.put(SCIMConstants.RoleSchemaConstants.DISPLAY_NAME, displayAttribute);
+        }
+    }
+
+    /**
+     * Deletes the current value of display name and rename it with the given value.
+     *
+     * @param displayName New display name.
+     */
+    public void replaceDisplayName(String displayName) {
+
+        replaceSimpleAttribute(SCIMSchemaDefinitions.SCIMRoleSchemaDefinition.DISPLAY_NAME, displayName);
+    }
+
+    /**
+     * Get the users of the role.
+     *
+     * @return List of users.
+     */
+    public List<String> getUsers() {
+
+        if (!this.isAttributeExist(SCIMConstants.RoleSchemaConstants.USERS)) {
+            return Collections.emptyList();
+        }
+        List<String> userList = new ArrayList<>();
+        MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                .get(SCIMConstants.RoleSchemaConstants.USERS);
+        List<Attribute> subValuesList = users.getAttributeValues();
+        for (Attribute subValue : subValuesList) {
+            ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
+            Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
+            if (subAttributesList == null ||
+                    !subAttributesList.containsKey(SCIMConstants.CommonSchemaConstants.VALUE)) {
+                continue;
+            }
+            userList.add((String) ((SimpleAttribute) (subAttributesList.get(
+                    SCIMConstants.CommonSchemaConstants.VALUE))).getValue());
+        }
+        return userList;
+    }
+
+    /**
+     * Get the users of the role with their display names.
+     *
+     * @return List of display names.
+     */
+    public List<String> getUsersWithDisplayName() {
+
+        if (!this.isAttributeExist(SCIMConstants.RoleSchemaConstants.USERS)) {
+            return Collections.emptyList();
+        }
+        MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                .get(SCIMConstants.RoleSchemaConstants.USERS);
+        List<Attribute> subValuesList = users.getAttributeValues();
+        if (subValuesList == null) {
+            return Collections.emptyList();
+        }
+        List<String> displayNames = new ArrayList<>();
+        for (Attribute subValue : subValuesList) {
+            ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
+            Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
+            if (subAttributesList == null ||
+                    !subAttributesList.containsKey(SCIMConstants.CommonSchemaConstants.DISPLAY)) {
+                continue;
+            }
+            displayNames.add((String) ((SimpleAttribute) (subAttributesList
+                    .get(SCIMConstants.CommonSchemaConstants.DISPLAY))).getValue());
+        }
+        return displayNames;
```

Comment:
Looks like the same logic is repeated in multiple get methods. Better write a private method that takes RoleSchemaConstants.X as input and returns the list.

Reply:
fixed in [25b26c2](https://github.com/wso2/charon/pull/392/commits/25b26c2228a54f30201b37c15e189987e55ccf32)